bin/publish

```diff
@@ -1,63 +1,87 @@
 #!/bin/bash -e
```

Comment:
I don't find the reorganization of this file to be more clear - in fact, it's more complicated imo, and jumps around quite a bit to follow the logic.

Some notes:
- For this project nothing is going to publish unless the condition `[ "$git_description" = "v${VERSION}" ]` is met, so there should be no call to a `publish_image` method outside of that condition check
- It's confusing to have `publish_image` and `publish_images` - those names are so similar, and on the surface I can't tell what is different between them except that perhaps one does the operation for more than one image
- If you really want to wrap it all up to be as simple as possible, why not do something like:
   ```
   # check that the tag matches the version
   # looping over the target images and tags (ie a double loop), publish to dockerhub
   # publish to redhat
   ```
   You can even set it up so that `publish_dockerhub_image` re-tags and pushes to dockerhub (that's really all it should need - you can pass in the base image, target, and tag) and `publish_redhat_image` does the whole login / push / error catching for a single image.
- Note: we can't push the RH FIPS image without creating an authn-k8s FIPS project in the RH container registry - so that should maybe get pushed to a follow-up ticket. We can leave the build steps in this PR, but I don't think we can publish that image yet in this PR
- I have a preference to bash methods being prefaced by `function`, but this is more of a nit - from the [style guide](https://google.github.io/styleguide/shellguide.html#function-names):
   > The function keyword is extraneous when “()” is present after the function name, but enhances quick identification of functions.
- There should be a space between the close parens of a method statement and the opening curly brace of the method definition, eg `main() {`